src/UnitTransformedLikelihood.cc

```diff
@@ -112,8 +111,8 @@ void setupArrays(
     bool usePixelWeights,
     double weightsMultiplier
 ) {
-    afw::detection::flattenArray(footprint, image.getImage()->getArray(), data, image.getXY0());
-    afw::detection::flattenArray(footprint, image.getVariance()->getArray(), variance, image.getXY0());
+    footprint.getSpans()->flatten(data, image.getImage()->getArray(), image.getXY0());
+    footprint.getSpans()->flatten(variance, image.getVariance()->getArray(), image.getXY0());
```

Comment:
This is a bit of code attempting to be as similar to the previous code as possible. It is also important to note that what you wrote above and what is said in the code are not exactly the same thing. What you wrote will create a new array and put it in the variable called data. The code as written will insert the data into an already existing array, which may be a view into a larger existing array.